Microsoft.Toolkit.Uwp.UI/Extensions/HyperlinkExtensions.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System.Windows.Input;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Documents;
+
+namespace Microsoft.Toolkit.Uwp.UI.Extensions
+{
+    /// <summary>
+    /// Provides attached dependency properties for the <see cref="Hyperlink"/> content element that allows
+    /// it to invoke a <see cref="ICommand"/> when clicked
+    /// </summary>
+    public static class HyperlinkExtensions
+    {
+        private static void OnCommandPropertyChanged(DependencyObject sender, DependencyPropertyChangedEventArgs args)
+        {
+            Hyperlink hyperlink = sender as Hyperlink;
+
+            if (hyperlink != null)
+            {
+                ICommand command = args.NewValue as ICommand;
+
+                if (command != null)
+                {
+                    hyperlink.Click += OnHyperlinkClicked;
+                }
+                else
+                {
+                    hyperlink.Click -= OnHyperlinkClicked;
+                }
+            }
+        }
+
+        private static void OnHyperlinkClicked(Hyperlink sender, HyperlinkClickEventArgs args)
+        {
+            ICommand command = GetCommand(sender);
+            object parameter = GetCommandParameter(sender);
+
+            if (command != null)
+            {
+                command.Execute(parameter);
+            }
+        }
+
+        /// <summary>
+        /// Attached <see cref="DependencyProperty"/> for binding an <see cref="ICommand"/> instance to a <see cref="Hyperlink"/>
+        /// </summary>
+        public static readonly DependencyProperty CommandProperty = DependencyProperty.RegisterAttached("Command", typeof(ICommand), typeof(HyperlinkExtensions), new PropertyMetadata(null, OnCommandPropertyChanged));
```

Comment:
Hi Scott,

Thanks for the incredibly fast feedback.

Re: using nameof(), these are attached properties so the property name specified is not a conventional property; i.e. "CommandProperty" attached dependency property exposes a "Command" property in XAML but there is no ```public ICommand Command { get; set; }``` property on the class. Therefore, I don't see how "nameof" can be used here.

Indeed, if I change the above line to:
```public static readonly DependencyProperty CommandProperty = DependencyProperty.RegisterAttached(nameof(Command), typeof(ICommand), typeof(HyperlinkExtensions), new PropertyMetadata(null, OnCommandPropertyChanged));```
I get - as expected - a compilation error: "The name 'Command' does not exist in the current context".

Obviously changing this to "nameof(CommandProperty)" is not desirable either so am I missing something or is this guideline not applicable to attached properties?

Thanks,
  Ian